model/flow/version_beacon.go

```diff
@@ -155,3 +156,43 @@ func (v *VersionBeacon) String() string {
 	}
 	return buffer.String()
 }
+
+// ProtocolStateVersionUpgrade is a service event emitted by the FlowServiceAccount
+// to signal an upgrade to the Protocol State version. `NewProtocolStateVersion`
+// must be strictly greater than the currently active Protocol State Version,
+// otherwise the service event is ignored.
+// If the node software supports `NewProtocolStateVersion`, then it uses the
+// specified Protocol State Version, beginning with the first block B where BOTH:
+//  1. The `ProtocolStateVersionUpgrade` service event has been sealed in B's ancestry
+//  2. B.view >= `ActiveView`
+//
+// NOTE: A ProtocolStateVersionUpgrade event `E` is only accepted while processing block `B`
+// which seals `E` if E.ActiveView < B.View + SafetyThreshold.
```

Comment:
Can you please take a look at the comparison. In my opinion, we need to enforce: "E.ActiveView **>** B.View + SafetyThreshold" while you are writing "E.ActiveView **<** B.View + SafetyThreshold"

```suggestion
// NOTE: A ProtocolStateVersionUpgrade event `E` is accepted while processing block `B`
// which seals `E` if and only if E.ActiveView > B.View + SafetyThreshold.
```

Reply:
Good eye, thank you 🙇